packages/devtools_app/lib/src/shared/config_specific/import_export/_export_web.dart

```diff
@@ -15,20 +16,37 @@ ExportControllerWeb createExportController() {
 class ExportControllerWeb extends ExportController {
   ExportControllerWeb() : super.impl();
 
-  @override
-  void saveFile({
-    required String content,
+  void _downloadJsObject({
+    // ignore: avoid-dynamic, there is not better way in this case
+    required dynamic content,
     required String fileName,
   }) {
     final element = document.createElement('a') as HTMLAnchorElement;
     element.setAttribute(
       'href',
-      URL.createObjectURL(Blob([content.toJS].toJS) as JSObject),
+      // ignore: avoid-dynamic, avoid_dynamic_calls, there is not better way in this case,
+      URL.createObjectURL(Blob(([content.toJS] as dynamic).toJS) as JSObject),
     );
     element.setAttribute('download', fileName);
     element.style.display = 'none';
     (document.body as HTMLBodyElement).append(element as JSAny);
     element.click();
     element.remove();
   }
+
+  @override
+  void saveFile({
+    required String content,
+    required String fileName,
+  }) {
+    _downloadJsObject(content: content, fileName: fileName);
+  }
+
+  @override
+  void saveDataFile({
```

Comment:
can these be merged into one method?
```
void saveFile<T>({
  required T content,
  required String fileName,
})
```

Reply:
+1. Can we not give `_downloadJsObject` a type argument as well to avoid the dynamic in the signature?